src/JsonReader.cs

```diff
@@ -548,127 +570,162 @@ sealed class NonProperty : IJsonProperty<Unit, JsonReadResult<Unit>>
-                {
-                    if (!property.IsMatch(ref reader))
-                        return false;
+                                values.CurrentPropertyIndex = index;
+                                return true;
+                            }
 
-                    if (!reader.Read())
-                        throw PartialJsonNotSupportedException();
+                                 if (TrySetPropertyIndex(1,  property1,  ref reader, ref values)) { }
+                            else if (TrySetPropertyIndex(2,  property2,  ref reader, ref values)) { }
+                            else if (TrySetPropertyIndex(3,  property3,  ref reader, ref values)) { }
+                            else if (TrySetPropertyIndex(4,  property4,  ref reader, ref values)) { }
+                            else if (TrySetPropertyIndex(5,  property5,  ref reader, ref values)) { }
+                            else if (TrySetPropertyIndex(6,  property6,  ref reader, ref values)) { }
+                            else if (TrySetPropertyIndex(7,  property7,  ref reader, ref values)) { }
+                            else if (TrySetPropertyIndex(8,  property8,  ref reader, ref values)) { }
+                            else if (TrySetPropertyIndex(9,  property9,  ref reader, ref values)) { }
+                            else if (TrySetPropertyIndex(10, property10, ref reader, ref values)) { }
+                            else if (TrySetPropertyIndex(11, property11, ref reader, ref values)) { }
+                            else if (TrySetPropertyIndex(12, property12, ref reader, ref values)) { }
+                            else if (TrySetPropertyIndex(13, property13, ref reader, ref values)) { }
+                            else if (TrySetPropertyIndex(14, property14, ref reader, ref values)) { }
+                            else if (TrySetPropertyIndex(15, property15, ref reader, ref values)) { }
+                            else if (TrySetPropertyIndex(16, property16, ref reader, ref values)) { }
+
+                            if (!reader.Read())
+                                return reader.Suspend((sm, values));
+
+                            sm.OnPropertyNameRead();
 
-                    switch (property.Reader.TryRead(ref reader))
-                    {
-                        case { Incomplete: true }:
-                            throw PartialJsonNotSupportedException();
-                        case { Error: { } err }:
-                            error = err;
-                            return false;
-                        case { Value: var val }:
-                            value = (true, val);
-                            return true;
+                            break;
+
+                        case ObjectReadStateMachine.ReadResult.PropertyValue:
+                            static JsonReadResult<TResult>? ReadPropertyValue<TValue>(IJsonProperty<TValue, JsonReadResult<TValue>> property,
```

Comment:
nit: Consider renaming `TValue` to simply `T`.

Reply:
Done in 0c09682f373f4e7535f978dacba5d24e8187ba9f